scripts/install_fast_mcljava_linux_mac.sh

```diff
@@ -38,28 +81,39 @@ java_inc=$1
   cd mcl
   git checkout $mcl_version || exit
   echo "----- Deleting currently installed version of mcl -----"
-  if [ $os == "linux" ]; then
+  if [ $os = "linux" ] || [ $os = "FreeBSD" ]; then
     sudo rm /usr/lib/libmcljava.so
   else # mac os
     mkdir -p ~/Library/Java/Extensions/ #check that this is included here: System.out.println(System.getProperty("java.library.path"));
 	  rm ~/Library/Java/Extensions/libmcljava.dylib
   fi
   echo "----- Building mcl -----"
-  make -j4 || exit # build mcl library
+  mkdir build 2>/dev/null
+  cd build
+  cmake .. -DCMAKE_C_FLAGS="${C_TUNING_FLAGS}" -DCMAKE_CXX_FLAGS="${C_TUNING_FLAGS}" -DMCL_STATIC_LIB="ON" 
+  cmake --build .
+  cd ..
+  export MCL_LIBDIR=$(pwd)/build/lib
   echo "----- Building mcl java bindings and running tests -----"
   echo "----- Java include path: $java_inc -----"
   cd ffi/java
-  make test_mcl JAVA_INC=-I$java_inc || exit # build java bindings, set include manually
+  mkdir build 2>/dev/null
+  cd build
+  export JAVA_HOME=$1/../
+  if [ -z ${2+x} ]; then
+	  cmake .. -DMCL_LINK_DIR=${MCL_LIBDIR} -DCMAKE_CXX_FLAGS="${C_TUNING_FLAGS} -I /usr/local/include"
+  else
+	  cmake .. -DMCL_LINK_DIR=${MCL_LIBDIR} -DCMAKE_CXX_FLAGS="${C_TUNING_FLAGS} -I /usr/local/include" -DGMP_LINK_DIR=/usr/local/lib
+  fi
+  cmake --build . -v
   echo "----- Copying mcl java shared library to /usr/lib/ -----"
-  cd ../..
-  if [ $os == "linux" ]; then
-    sudo cp lib/libmcljava.so /usr/lib/
+  if [ $os = "linux" ] || [ $os = "FreeBSD" ]; then
+    sudo cp libmcljava.so /usr/lib/
   else # mac os
     mkdir -p ~/Library/Java/Extensions/ #check that this is included here: System.out.println(System.getProperty("java.library.path"));
-	  cp lib/libmcljava.dylib ~/Library/Java/Extensions/
+	  cp libmcljava.dylib ~/Library/Java/Extensions/
   fi
   echo "----- Installation finished successfully. Deleting mcl repository folder -----"
-  cd ..
-  rm -rf mcl
+  rm -r /tmp/mcl
```

Comment:
Can you make this `-rf` again? This version tends to not delete some read-only files in the `.git` dir.